misc/collect_logs.sh

```diff
@@ -0,0 +1,293 @@
+  info "name of 'mvi.logs.<timestamp>'."
+  info ""
+  info "Options:"
+  info "  -h, --help          Display this help text."
+  info "      --namespace     The namespace that MVI is installed in. If not set,"
+  info "                      will use the default namespace."
+  info "      --standalone    Will collect OS information from the system this script"
+  info "                      is running on. Only set if you are running MVI in"
+  info "                      standalone mode on this system."
+}
+
+# Logging functions
+info () {
+  # Green
+  printf "\E[0;32m[ INFO ] "
+  tput sgr0;
+  printf "%s\n" "$@";
+}
+warning () {
+  # Yellow
+  printf "\E[0;33m[ WARN ] "
+  tput sgr0;
+  printf "%s\n" "$@";
+}
+error () {
+  # Red
+  printf "\E[0;31m[ FAIL ] "
+  tput sgr0;
+  printf "%s\n" "$@";
+}
+
+# Helper function for waiting for processes
+check_pid()
+{
+  local cnt=0
+  while [[ $cnt -lt 5 ]]
+  do
+    sleep 1
+    if [[ ! $(ps -p $1) =~ $1 ]]; then break; fi
+    cnt=$((cnt+1))
+    info "cxlffdc: waiting for slow process $1"
+  done
+  if [[ $(ps -p $1) =~ $1 ]]
+  then
+    kill -9 $1
+  fi
+}
+
+# Make sure kubectl or oc is set up correctly
+if command -v kubectl > /dev/null
+then
+  KUBE_CMD="kubectl"
+elif command -v oc > /dev/null
+then
+  KUBE_CMD="oc"
+else
+  error "Either 'kubectl' or 'oc' is required to collect logs, but neither could be"
+  error "found in the path.You can find installation instructions at:"
+  error "  https://docs.openshift.com/container-platform/4.3/cli_reference/openshift_cli/getting-started-cli.html"
+  exit 1
+fi
+
+# Process arguments
+while [ $# -gt 0 ]
+do
+  key="$1"
+
+  case $key in
+    -n|--namespace)
+    NAMESPACE="$2"
+    shift
+    shift
+    ;;
+    -s|--standalone)
+    STANDALONE=true
+    shift
+    ;;
+    -v|--version)
+    info "v1.1"
+    exit 0
+    ;;
+    -h|--help)
+    usage
+    exit 0
+    ;;
+    *) # Unknown option
+    echo "Unknown option: $key"
+    usage
+    exit 1
+    ;;
+  esac
+done
+
+KUBE_CMD="${KUBE_CMD} --namespace ${NAMESPACE}"
+info "Using kube/oc command '${KUBE_CMD}'"
+
+# Set up the directory that we'll be dumping the logs into
+if [[ -e $DATADIR ]]; then rm -rf $DATADIR 2>/dev/null; fi
+mkdir -p $DATADIR
+
+# Start actually collecting logs
+info "Collecting ${PRODUCT_NAME} logs in parallel..."
+
+# K8s/OCP information
+$KUBE_CMD get clusterissuer -o wide > $DATADIR/get_clusterissuer.txt &
+$KUBE_CMD version > $DATADIR/version.txt &
+$KUBE_CMD get nodes -o wide > $DATADIR/get_nodes.txt &
+$KUBE_CMD describe nodes > $DATADIR/describe_nodes.txt &
+$KUBE_CMD get ingress -o yaml > $DATADIR/get_ingress.yaml &
+$KUBE_CMD get componentstatus -o wide > $DATADIR/get_componentstatus.txt &
+$KUBE_CMD api-versions > $DATADIR/kubectl_apiversions.txt &
+
+$KUBE_CMD get all -o yaml > $DATADIR/get_all.yaml &
+# The all-namespaces flag takes precedence over --namespace
+$KUBE_CMD get events --all-namespaces > $DATADIR/get_all_events.txt &
```

Comment:
You set `$KUBE_CMD` to have `--namespace` flag. Now you are adding `--all-namespaces`, is that going to work as you expect it to?

Reply:
Yes it does. The `--all-namespaces` flag ignores the `--namespace` one.